openstates/wi/bills.py

```diff
@@ -329,36 +329,29 @@ def add_senate_votes(self, vote, url):
             return
 
         doc = lxml.html.fromstring(html)
+        trs = doc.xpath('//table[@class="senate"]/tbody/tr')
 
-        yes_count = no_count = other_count = 0
+        vote_types = ['yes', 'no', 'other']
+        vote_counts = {}  # Vote counts for yes, no, other
+        name_counts = {}
 
-        # a game of div-div-table
-        for ddt in doc.xpath('//div/div/table'):
-            text = ddt.text_content()
-            if 'Wisconsin Senate' in text or 'SEQUENCE NO' in text:
-                continue
-            elif 'AYES -' in text:
-                for name in text.split('\n\n\n\n\n')[1:]:
-                    if name.strip() and 'AYES' not in name:
-                        vote.vote('yes', name.strip())
-                        yes_count += 1
-            elif 'NAYS -' in text:
-                for name in text.split('\n\n\n\n\n')[1:]:
-                    if name.strip() and 'NAYS' not in name:
-                        vote.vote('no', name.strip())
-                        no_count += 1
-            elif 'NOT VOTING -' in text:
-                for name in text.split('\n\n\n\n\n')[1:]:
-                    if name.strip() and "NOT VOTING" not in name:
-                        vote.vote('other', name.strip())
-                        other_count += 1
-            elif text.strip():
-                raise ValueError('unexpected block in vote')
-
-        if yes_count or no_count or other_count:
-            vote.set_count('yes', yes_count)
-            vote.set_count('no', no_count)
-            vote.set_count('other', other_count)
+        for index in range(0, 5, 2):
```

Comment:
This seems like weird construction, both non-standard Python and hard to read (for future maintainers). Could you do something like this instead?

- XPath for `<td>` that has text of `(AYES|NAYS|NOT VOTING) - \d+`
- get the names from the `<table>` in the [following-sibling](https://www.w3schools.com/xml/xpath_axes.asp) `<td>`

Reply:
100% agreed, doing it.